### DISCUSSION TITLE
Issue with LDAP specified in ldap2pg.yml

### DISCUSSION THREAD

**Discussion by JBHighHair** [dalibo/ldap2pg#671] [Q&A] [answered]
Hi,

I've a PostgreSQL v16.4 server running on Windows server.
SSPI is configured correctly in pg_hba.config and works as I can use pgAdmin to connect to the server from a remote PC.
I'm using Windows version of ldap2pg version 6.2 (ldap2pg_6.2_windows_amd64.tar) to try to synch specific Active Directory group members.
When performing a dry run, the LDAP server is localhost rather than somedomain.com...

```
time=2024-11-08T15:04:38.015+11:00 level=DEBUG msg="LDAP dial." uri=ldap://localhost try=1
time=2024-11-08T15:04:38.018+11:00 level=DEBUG msg=Retrying. err="LDAP Result Code 200 \"Network Error\": dial tcp [::1]:389: connectex: No connection could be made because the target machine actively refused it." attempt=0
...
time=2024-11-08T15:04:38.015+11:00 level=DEBUG msg="LDAP dial." uri=ldap://localhost try=10
time=2024-11-08T15:04:38.018+11:00 level=DEBUG msg=Retrying. err="LDAP Result Code 200 \"Network Error\": dial tcp [::1]:389: connectex: No connection could be made because the target machine actively refused it." attempt=9
```
My ldap2pg.yml file contains ldap section....

```
ldap:
  binddn: CN=ldaplookupacct,OU=Delegation,OU=Users,DC=somedomain,DC=com
  password: password
  uri: ldap://somedomain.com
```

localhost is not specified in any part of the ldap2pg.yml file.

On the same cluster, I'm able to successfully use the same LDAP server via pg_hba.conf to authenticate some users.

```
host    all            all             0.0.0.0/0                ldap       ldapserver=somedomain.com ldapport=389 ldapbinddn="CN=ldaplookupacct,OU=Delegation,OU=Users,DC=somedomain,DC=com" ldapbindpasswd="somepassword" ldapbasedn="DC=somedomain,DC=com" ldapsearchattribute="sAMAccountName"
```

I've tested with version 6.1 which produces the same problem, localhost is displayed.

However, with version 6.2 if you use environment variables it works....
set LDAPURI=ldap://somedomain.com
set LDAPBASE=DC=somedomain,DC=com
set LDAPBINDDN=CN=ldaplookupacct,OU=Delegation,OU=Users,DC=somedomain,DC=com
set LDAPPASSWORD=password

Does anyone know why it goes straight to localhost and ignores my LDAP declaration in ldap2pg.yml ?

Thanks.

**Comment by bersace** [comment, marked as answer]
Hi @JBHighHair , I dropped `ldap:` top level section as of 6.0. This way config file has no secret and we have less confusion between top level `ldap:` and `ldap:` rules (renamed as `ldapsearch:`).

See reference configuration: https://ldap2pg.readthedocs.io/en/latest/config/

Configure ldap2pg with libldap files and .env file. See https://ldap2pg.readthedocs.io/en/latest/cli/#environment-variables

Regards.

**Comment by JBHighHair** [comment]
Thank you.
